design_drafts/components_architecture_and_urdf_examples.md

```diff
@@ -0,0 +1,424 @@
+      <param name="max_velocity_value">1</param>
+      <param name="min_effort_value">-0.5</param>
+      <param name="max_effort_value">0.5</param>
+    </joint>
+  </ros2_control>
+```
+
+#### 4. Industrial Robots with a sensor integrated into the robot's control box
+  * the communication is done using proprietary API
+  * Data for all joints is exchanged in batch (at once)
+  * Sensor data are exchanged together with joint data
+  * Examples: KUKA RSI with sensor connected to KRC (KUKA control box)
+
+```xml
+  <ros2_control name="2DOF_System_Robot_with_Sensor" type="system">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_Sensor</classType>
+      <param name="example_param_write_for_sec">2</param>
+      <param name="example_param_read_for_sec">2</param>
+    </hardware>
+    <joint name="joint1">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="can_read">True</param>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+    <joint name="joint2">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="can_read">True</param>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+    <sensor name="tcp_fts_sensor">
+      <classType>ros2_control_components/ForceTorqueSensor</classType>
+      <param name="frame_id">kuka_tcp</frame_id>
+      <param name="lower_limits">-100</param>
+      <param name="upper_limits">100</param>
+    </sensor>
+  </ros2_control>
+```
+Note:
+  * `ros2_control_components/PositionJoint`type has implicitly:
+    ```xml
+      <commandInterfaceType>position</commandInterfaceType>
+      <stateInterfaceType>position</stateInterfaceType>
+    ```
+
+
+#### 5. Industrial Robots with a sensor connected to ROS computer
+  * the communication is done using proprietary API
+  * Data for all joints is exchanged at once
+  * Sensor data are exchanged independently
+  * Examples: KUKA RSI and FTS connected to ROS-PC
+
+```xml
+  <ros2_control name="2DOF_System_Robot_Position_Only_External_Sensor" type="system">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_Position_Only</classType>
+      <param name="example_param_write_for_sec">2</param>
+      <param name="example_param_read_for_sec">2</param>
+    </hardware>
+    <joint name="joint1">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="can_read">True</param>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+    <joint name="joint2">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="can_read">True</param>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+  </ros2_control>
+  <ros2_control name="2DOF_System_Robot_ForceTorqueSensor" type="sensor">
```

Comment:
I think it deserves a bit more context or description that we have two separate `ros2_control` tags in here and thus treat the two components individually.